.github/workflows/go-update-commenter.yml

```diff
@@ -0,0 +1,56 @@
+name: "Go update commenter"
+
+on:
+  pull_request:
+    # Only run on PR label events (in particular not on every commit)
+    types: [ labeled ]
+
+jobs:
+  old-versions-match:
+    # Only run if the PR is labeled with 'go-update'
+    if: ${{ github.event.label.name == 'go-update' }}
+    runs-on: ubuntu-latest
+    steps:
+      # get the Go version of the target branch
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.base_ref }}
+      - name: Get former Go version
+        id: former_go_version
+        run: |
+          echo version="$(cat .go-version)" >> $GITHUB_OUTPUT
+
+      # get the Go version of the PR branch
+      - uses: actions/checkout@v3
+      - name: Get current Go version
+        id: new_go_version
+        run: |
+          echo version="$(cat .go-version)" >> $GITHUB_OUTPUT
+
+      # build the comment
+      - name: Build full comment
+        id: old_versions
+        run: |
+          set -euxo pipefail
+          GITHUB_HEAD_URL='${{ github.server_url }}/${{ github.repository }}/blob/${{ github.sha }}'
+          {
+            echo "matches<<EOF"
+            echo "Here are potential matches of the former version:"
+            echo ""
+            bash -x ./tools/go-update/detect-old-version.sh "${{ steps.former_go_version.outputs.version }}" "${{ steps.new_go_version.outputs.version }}" | \
+              sed -E 's|^([^:]+):([^:]+):\s*(\S.*)$|- [\1:\2]('"$GITHUB_HEAD_URL"'/\1#L\2): `\3`|'
```

Comment:
Could we move this line to a `./tools/go-update/update-old-version.sh` ? Or add an option to the detect script to do the update. Having the `sed` in a script would be better and less error prone.

Reply:
The `sed` is simply for changing the output of grep to have URLs instead of just having a file path, it doesn't edit any file.
We need some PR-specific information (at least the commit SHA) to build the Github URL, which is why it is in the job definition itself.

Most matches are expected to be false positives (known true positives are supposed to be changed by the invoke task), this job is 100% to have potential missed matches appear clearly on the PR, but it doesn't do anything except write a comment

Let me know if you still think it should be moved in a separate script !

Reply:
Sorry, I miss understood the `sed`. Thanks for the explanation !

Reply:
No worries, if the code / script isn't clear enough it means I need to improve or add comments !